src/esp/metadata/managers/StageAttributesManager.cpp

```diff
@@ -191,27 +191,51 @@ StageAttributes::ptr StageAttributesManager::initNewObjectInternal(
     newAttributes->setRenderAssetHandle(attributesHandle);
     newAttributes->setCollisionAssetHandle(attributesHandle);
 
-    // set defaults for navmesh default handles and semantic mesh default
-    // handles
-    std::string navmeshFilename =
-        io::changeExtension(attributesHandle, ".navmesh");
-
-    if (Corrade::Utility::Directory::exists(navmeshFilename)) {
-      newAttributes->setNavmeshAssetHandle(navmeshFilename);
-    }
-    // Build default semantic descriptor file name
-    std::string houseFilename = io::changeExtension(attributesHandle, ".house");
-
-    if (!Corrade::Utility::Directory::exists(houseFilename)) {
-      houseFilename = io::changeExtension(attributesHandle, ".scn");
-    }
-    newAttributes->setHouseFilename(houseFilename);
-
-    // Build default semantic mesh file name
-    const std::string semanticMeshFilename =
-        Cr::Utility::Directory::splitExtension(houseFilename).first +
-        "_semantic.ply";
-    newAttributes->setSemanticAssetHandle(semanticMeshFilename);
+    if (attributesHandle != "NONE") {
+      // TODO when all datasets have configuration support, get rid of all these
+      // default settings
+
+      // set defaults for navmesh, semantic mesh and lexicon handles
+      // start with root stage name, including path but without final extension
+      // default navmesh should have .navmesh ext
+      std::string navmeshFilename =
+          this->findFilenameUsingCriteria(attributesHandle, {".navmesh"});
+      // if present, file was found, so set value (overriding attributes
+      // default)
+      if (!navmeshFilename.empty()) {
+        newAttributes->setNavmeshAssetHandle(navmeshFilename);
```

Comment:
Don't forget to move temp variable strings. 👍 
```suggestion
        newAttributes->setNavmeshAssetHandle(std::move(navmeshFilename));
```